File: tests/benchmarks/http/bench_http_simple.go
package main

import (
	"net/http"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
		w.Write([]byte("Hello, World!"))
	})

	http.HandleFunc("/json", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json; charset=utf-8")
		w.Write([]byte(`{"message":"Hello, World!"}`))
	})

	http.ListenAndServe(":8081", nil)
}

File: tests/benchmarks/http/bench_http_simple.js
const http = require('http');

const server = http.createServer((req, res) => {
    if (req.url === '/json') {
        res.writeHead(200, { 'Content-Type': 'application/json; charset=utf-8' });
        res.end('{"message":"Hello, World!"}');
    } else {
        res.writeHead(200, { 'Content-Type': 'text/plain; charset=utf-8' });
        res.end('Hello, World!');
    }
});

server.listen(8082, () => {
    console.log('Node.js listening on :8082');
});
